src/sqlfluff/core/parser/segments/base.py

```diff
@@ -1268,6 +1268,7 @@ def iter_patches(
             # This segment isn't a literal, but has changed, we need to go deeper.
 
             # Iterate through the child segments
+            source_idx = self.pos_marker.source_slice.start
```

Comment:
Note that `source_idx` is computed the same way as `templated_idx`, but using `source_slice` rather than `templated_slice`.